src/plugins/event-plugins/NavigationPlugin.ts

```diff
@@ -47,6 +57,24 @@ export class NavigationPlugin implements Plugin {
         }
     }
 
+    /**
+     * Use the loadEventEnd field from window.performance to check if the website
+     * has loaded already.
+     * @returns boolean
+     */
+    hasTheWindowLoadEventFired() {
+        if (
+            window.performance &&
+            window.performance.getEntriesByType(NAVIGATION).length
+        ) {
```

Comment:
Does this guarantee that `window.load` has fired? 

Or should we look for a specific event that signals the page has finished loading -- i.e., [`PerformanceTiming.loadEventEnd`](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceTiming/loadEventEnd) or [`PerformanceNavigationTiming.loadEventEnd`](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming/loadEventEnd)

Reply:
Discussed offline; The following line to the if statement uses PerformanceNavigationTiming.loadEventEnd